database-commons/src/main/java/io/cdap/plugin/db/connector/AbstractDBSpecificConnectorConfig.java

```diff
@@ -0,0 +1,130 @@
+ */
+public abstract class AbstractDBSpecificConnectorConfig extends PluginConfig implements DBConnectorProperties {
+
+  public static final String USER = "user";
+  public static final String HOST = "host";
+  public static final String PORT = "port";
+  public static final String PASSWORD = "password";
+  public static final String CONNECTION_ARGUMENTS = "connectionArguments";
+  public static final String JDBC_PLUGIN_NAME = "jdbcPluginName";
+
+  @Name(HOST)
+  @Description("Database host")
+  @Macro
+  protected String host;
+
+  @Name(PORT)
+  @Description("Database port number")
+  @Macro
+  @Nullable
+  protected Integer port;
+
+  @Name(JDBC_PLUGIN_NAME)
+  @Description("Name of the JDBC driver to use. This is the value of the 'jdbcPluginName' key defined in the JSON " +
+    "file for the JDBC plugin.")
+  protected String jdbcPluginName;
+
+  @Name(USER)
+  @Description("User to use to connect to the specified database. Required for databases that " +
+    "need authentication. Optional for databases that do not require authentication.")
+  @Nullable
+  @Macro
+  protected String user;
+
+  @Name(PASSWORD)
+  @Description("Password to use to connect to the specified database. Required for databases that " +
+    "need authentication. Optional for databases that do not require authentication.")
+  @Nullable
+  @Macro
+  protected String password;
+
+  @Name(CONNECTION_ARGUMENTS)
+  @Description("A list of arbitrary string key/value pairs as connection arguments.")
+  @Nullable
+  @Macro
+  protected String connectionArguments;
+
+  @Nullable
+  @Override
+  public String getUser() {
+    return user;
+  }
+
+  @Nullable
+  @Override
+  public String getPassword() {
+    return password;
+  }
+
+  @Override
+  public Properties getConnectionArgumentsProperties() {
+    return getConnectionArgumentsProperties(connectionArguments, user, password);
+  }
+
+  @Override
+  public String getJdbcPluginName() {
+    return jdbcPluginName;
+  }
+
+  protected static Properties getConnectionArgumentsProperties(@Nullable String connectionArguments,
+                                                            @Nullable String user, @Nullable String password) {
+    KeyValueListParser kvParser = new KeyValueListParser("\\s*;\\s*", "=");
+
+    Map<String, String> connectionArgumentsMap = new HashMap<>();
+    if (!Strings.isNullOrEmpty(connectionArguments)) {
+      for (KeyValue<String, String> keyVal : kvParser.parse(connectionArguments)) {
+        connectionArgumentsMap.put(keyVal.getKey(), keyVal.getValue());
+      }
+    }
+
+    if (user != null) {
+      connectionArgumentsMap.put("user", user);
+    }
+    if (password != null) {
+      connectionArgumentsMap.put("password", password);
+    }
+    Properties properties = new Properties();
+    properties.putAll(connectionArgumentsMap);
+    return properties;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public abstract int getPort();
```

Comment:
Could you handle default value in some other way? E.g. with a separate abstract `getDefaultPort`. It's a bit confusing to see a field and an abstract getter.

Reply:
Good idea :)

Reply:
done:)